monai/data/meta_tensor.py

```diff
@@ -253,6 +253,8 @@ def __torch_function__(cls, func, types, args=(), kwargs=None) -> Any:
         # we might have 1 or multiple outputs. Might be MetaTensor, might be something
         # else (e.g., `__repr__` returns a string).
         # Convert to list (if necessary), process, and at end remove list if one was added.
+        if isinstance(ret, torch.Size):  # tensor.shape/tensor.size()
```

Comment:
so this works, but is only a workaround for torch.Size,  there might be other bugs of other Tensor return types.

I think the culprit of the problem is update_meta() function
https://github.com/Project-MONAI/MONAI/blob/82852d89e950d1583f8f787996e71ba9aec8d1c1/monai/data/meta_tensor.py#L276

https://github.com/Project-MONAI/MONAI/blob/82852d89e950d1583f8f787996e71ba9aec8d1c1/monai/data/meta_tensor.py#L242

`isinstance(rets, tuple)` returns True for torch.Size (since it's a subclass of tuple) and then, we explicitely re-cast it to tuple()

we could change this line to 
`return type(rets)(out) if isinstance(rets, tuple) else out`

to preserve the exact sub-class type (such as torch.Size),  

but overall I don't understand why we modify the return types of at all, and check for tuples there,  I dont have a good understanding of this function.  so perhaps there is a better , more general fix

Reply:
The idea here is that if a torch function returns a torch.Size already, there's no need to try to attach meta data to the return value.

The tuple check and creation you mentioned is indeed suspicious, probably it will cause bugs for the ret if it's a tuple subclass such as namedtuple. But I'm not aware of any function to trigger this bug, maybe it's ok to keep it simple for now.

Reply:
so I'm okay we your fix as is, this solves my particular issue.  But I agree with your conclusion that this Bug will reappear again for some other return type.

I checked `MetaTensor.update_meta` and it's only applicable for a list of MetaTensors, so may be we can simply check if ret is not MetaTensor or not a sequence of MetaTensors then return ret right away?